source/system_stm32f4xx.c

```diff
@@ -81,8 +81,12 @@
 #include "stm32f4xx.h"
 #include "hal_tick.h"
 
-#if !defined  (HSE_VALUE) 
-  #define HSE_VALUE    ((uint32_t)8000000) /*!< Default value of the External oscillator in Hz */
+#ifndef YOTTA_CFG_HARDWARE_EXTERNALCLOCK
+#define YOTTA_CFG_HARDWARE_EXTERNALCLOCK (8000000)
+#endif
+
+#if !defined  (HSE_VALUE)
+  #define HSE_VALUE    ((uint32_t)(YOTTA_CFG_HARDWARE_EXTERNALCLOCK)) /*!< Default value of the External oscillator in Hz */
```

Comment:
Either:

1. Create an error if both (`YOTTA_CFG_HARDWARE_EXTERNALCLOCK` and `HSE_VALUE`) are defined
2. Create an error if `HSE_VALUE` is defined, regardless of `YOTTA_CFG_HARDWARE_EXTERNALCLOCK`

I vote for 2.